### PR TITLE
Split Italy

### PR DESCRIPTION
**Pull request by corradio**
Ref https://github.com/tmrowco/electricitymap-contrib/issues/1029
Superseeds https://github.com/tmrowco/electricitymap-contrib/pull/1536

Needs to be done:
- [x] Add geometries
- [x] Add ENTSOE exchange mappings
- [x] Add ENTSOE production mappings
- [x] Add bounding boxes

### CODE REVIEW

**Comment by corradio** [comment]
This is what it looks like for now:

untitled9

**Comment by alixunderplatz** [comment]
@corradio Looks very good so far!

**Comment by corradio** [comment]
@alixunderplatz what do you think about both SACOAC/SACODC <-> IT-SAR: should we sum them?

**Comment by brunolajoie** [comment]
Can't wait to see it live. We should also investigate what is the yearly average average "unknown" fuel mix in each of these region to refine carbon intensities!

**Comment by alixunderplatz** [comment]
@corradio regarding your first question: Maybe we get the everything on the mainland right, and then continue with Sardinia in another issue/PR. The grid infrastructure is a bit complicated and I can't quickly answer your question before the weekend.
A part of the energy in the DC-line between IT-CNO and Sardinia could be leaving the line in Corsica, so the amount reaching Sardinia may be a bit smaller. [See #1029](https://github.com/tmrowco/electricitymap-contrib/issues/1029#issuecomment-373960398)
I am not sure how to draw the exchange arrows, too.
If somebody could check all the exchanges on ENTSO-E related to Sardinia / Corsica, this would be terrific.

**Comment by alixunderplatz** [comment]
Just a quick summary regarding Sardinia for what you see in "Cross-Border physical flow" on ENTSO-E selecting "border-bidding zone":

"IT-Centre-North-IT-SACODC" is the exchange between IT-CNO and the DC-cable to Sardinia/Corsica.
"IT-SACODC-IT-Sardinia" is the exchange with the end of the DC-cable and Sardinia (the real exchange IT-CNO <-> Sardinia).
The difference is leaving the line on Corsica (IT-CNO <-> Corsica).
"IT-SACOAC-IT-Sardinia" is an AC-exchange Sardinia <-> Corsica.
Not sure how often the case appears that energy is fed from Sardinia towards IT-CNO and a part is leaving in Corsica. This is what "worries" me the most, although the max. power ever leaving in Corsica should be ~50 MW.

Carbon intensity on Corsica will remain unknown due to a lack of generation data.

Maybe handling SACODC as pseudo-zone for the map will make things a bit easier? Sorry if this complicates matters even more :/

**Comment by corradio** [comment]
@alixunderplatz I've left out the SACODC/AC for now.
Please review very carefully and check I've added all imports/exports and properly summed up the zones.

File: .circleci/config.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ stages:
           flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics
           # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
           flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
-          pylint -E parsers/*.py
+          pylint -E parsers/*.py -d unsubscriptable-object,unsupported-assignment-operation
```
[the remark this answers is not in the document]
Comment (corradio) [reply .circleci/config.yml]: Because of this: https://github.com/PyCQA/pylint/issues/1498

**Comment by maxbellec** [comment]
I found `_sum_production` hard to understand so I rewrote it. If you prefer the original version, we can roll back.

**Comment by corradio** [comment]
@maxbellec did you test that it returns exactly the same results? It's counter-productive if I now have to review your changes

**Comment by maxbellec** [comment]
yes the ouput is exactly the same for `fetch_production('IT-SIC')`

**Comment by corradio** [comment]
perfect! Thanks for cleaning up :)

**Comment by maxbellec** [comment]
Sorry, I know generally it's better to comment than change directly but I thought the entire function had to be re-written if we wanted something clean.
Just to check: when merging productions from zone A and B, we only keep datapoints whose datetime are in A production **and** B production, right? Otherwise we'll end up with wrong data I believe.

File: web/generate-zone-bounding-boxes.js [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
+
 function readNDJSON(path) {
   return fs.readFileSync(path, 'utf8').split('\n')
     .filter(d => d !== '')
     .map(JSON.parse);
 }
 
 let zones = readNDJSON('build/zonegeometries.json');
+if (args.length > 0) {
+  zones = zones.filter(d => d.properties.zoneName === args[0]);
```
Comment (maxbellec) [inline web/generate-zone-bounding-boxes.js]: :+1:

**Comment by maxbellec** [comment]
haven't checked the mappings yet though

**Comment by corradio** [comment]
yes that's right.

**Comment by alixunderplatz** [comment]
@corradio micro-zones are summed correctly with the larger zones. All exchanges are included 
👍 👍 👍

**Comment by corradio** [comment]
@alixunderplatz great! Can you approve the PR then?

**Pull request merged**